tests/distinct_on_fields/tests.py

```diff
@@ -70,7 +71,8 @@ def test_basic_distinct_on(self):
             ),
             (
                 Tag.objects.order_by('parent__pk', 'pk').distinct('parent'),
-                ['<Tag: t2>', '<Tag: t4>', '<Tag: t1>'],
+                ['<Tag: t2>', '<Tag: t4>', '<Tag: t1>'] if connection.features.nulls_order_largest
+                else ['<Tag: t1>', '<Tag: t2>', '<Tag: t4>'],
```

Comment:
Could we add an explicit `nulls_last` or `nulls_first` here instead?

Reply:
Perhaps not yet. CockroachDB doesn't support NULL FIRST/LAST modifiers (https://github.com/cockroachdb/cockroach/issues/6224) so `DatabaseFeatures.supports_order_by_nulls_modifier = False` which means the `IS NOT NULL` workaround is used.

`Tag.objects.order_by(F('parent__pk').asc(nulls_first=True), F('pk')).distinct('parent')` generates:
```sql
SELECT DISTINCT ON ("distinct_on_fields_tag"."parent_id") "distinct_on_fields_tag"."id", "distinct_on_fields_tag"."name", "distinct_on_fields_tag"."parent_id"
FROM "distinct_on_fields_tag"
ORDER BY "distinct_on_fields_tag"."parent_id" IS NOT NULL, "distinct_on_fields_tag"."parent_id" ASC, "distinct_on_fields_tag"."id" ASC;
```
which gives
```
django.db.utils.ProgrammingError: SELECT DISTINCT ON expressions must match initial ORDER BY expressions
```
The same queryset works on PostgreSQL.